File: archived/Calendar/js/js/scenario1-data.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var Globalization = Windows.Globalization;
    var Calendar = Globalization.Calendar;
    var CalendarIdentifiers = Globalization.CalendarIdentifiers;
    var ClockIdentifiers = Globalization.ClockIdentifiers;

    var outputText;

    var page = WinJS.UI.Pages.define("/html/scenario1-data.html", {
        ready: function (element, options) {
            document.getElementById("showResults").addEventListener("click", showResults);
            outputText = document.getElementById("outputText");
        }
    });

    function reportCalendarData(calendar, calendarLabel) {
        var results = calendarLabel + ": " + calendar.getCalendarSystem() + "\n";
        results += "Name of Month: " + calendar.monthAsSoloString() + "\n";
        results += "Day of Month: " + calendar.dayAsPaddedString(2) + "\n";
        results += "Day of Week: " + calendar.dayOfWeekAsSoloString() + "\n";
        results += "Year: " + calendar.yearAsString() + "\n";
        results += "\n";
        return results;
    }

    function showResults() {
        // This scenario uses the Windows.Globalization.Calendar class to display the parts of a date.

        // Create Calendar objects using different constructors.
        var calendar = new Calendar();
        var japaneseCalendar = new Calendar(["ja-JP"], CalendarIdentifiers.japanese, ClockIdentifiers.twelveHour);
        var hebrewCalendar = new Calendar(["he-IL"], CalendarIdentifiers.hebrew, ClockIdentifiers.twentyFourHour);

        // Display the results
        outputText.innerText = 
            reportCalendarData(calendar, "User's default calendar system") +
            reportCalendarData(japaneseCalendar, "Calendar system") +
            reportCalendarData(hebrewCalendar, "Calendar system");
    }
})();

File: Samples/Calendar/cppwinrt/Scenario1_Data.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1_Data.h"
#include "Scenario1_Data.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Globalization;
using namespace Windows::UI::Xaml;

namespace
{
    void ReportCalendarData(std::wostream& results, Calendar const& calendar, PCWSTR calendarLabel)
    {
        results << calendarLabel << L": " << std::wstring_view(calendar.GetCalendarSystem()) << L"\n" <<
        L"Name of Month: " << std::wstring_view(calendar.MonthAsSoloString()) << L"\n" <<
        L"Day of Month: " << std::wstring_view(calendar.DayAsPaddedString(2)) << L"\n" <<
        L"Day of Week: " << std::wstring_view(calendar.DayOfWeekAsSoloString()) << L"\n" <<
        L"Year: " << std::wstring_view(calendar.YearAsString()) << L"\n\n";
    }
}

namespace winrt::SDKTemplate::implementation
{
    Scenario1_Data::Scenario1_Data()
    {
        InitializeComponent();
    }

    void Scenario1_Data::ShowResults_Click(IInspectable const&, RoutedEventArgs const&)
    {
        // This scenario uses the Windows.Globalization.Calendar class to display the parts of a date.

        // Create Calendar objects using different constructors.
        Calendar calendar;
        Calendar japaneseCalendar({ L"ja-JP" }, CalendarIdentifiers::Japanese(), ClockIdentifiers::TwelveHour());
        Calendar hebrewCalendar({ L"he-IL" }, CalendarIdentifiers::Hebrew(), ClockIdentifiers::TwentyFourHour());

        // Generate the results
        std::wostringstream results;
        ReportCalendarData(results, calendar, L"User's default calendar system");
        ReportCalendarData(results, japaneseCalendar, L"Calendar system");
        ReportCalendarData(results, hebrewCalendar, L"Calendar system");
        OutputTextBlock().Text(results.str());
    }
}
